agilent_infiniium_scope.py

```diff
@@ -31,446 +31,344 @@
-        probeAtten = T.Value(float(probeAtten),'')
-        termination = T.Value(float(termination),'')
-        scale = T.Value(float(scale),'')
-        position = T.Value(float(position),'')
-        coupling = coupling
-        bwLimit = T.Value(float(bwLimit),'')
-        invert = invert
-        unit = unit[1:-1] #Get's rid of an extra set of quotation marks
-
-        returnValue((probeAtten,termination,scale,position,coupling,bwLimit,invert,unit))
-
-    @setting(111, channel = 'i', coupling = 's', returns=['s'])
-    def couplingNONEXISTANT(self, c, channel, coupling = None):
-        """Get or set the coupling of a specified channel
-        Coupling can be "AC", "DC", or "GND"
-        """
-        raise Exception('Not yet implemented')
-        dev = self.selectedDevice(c)
-        if coupling is None:
-            resp = yield dev.query('CH%d:COUP?' %channel)
-        else:
-            coupling = coupling.upper()
-            if coupling not in COUPLINGS:
-                raise Exception('Coupling must be "AC", "DC", or "GND"')
-            else:
-                yield dev.write(('CH%d:COUP '+coupling) %channel)
-                resp = yield dev.query('CH%d:COUP?' %channel)
-        returnValue(resp)
-
-    @setting(112, channel = 'i', scale = 'v', returns = ['v'])
-    def scale(self, c, channel, scale = None):
+    @setting(112, channel='i', scale='v', returns=['v'])
+    def scale(self, c, channel, scale=None):
         """Get or set the vertical scale of a channel in voltage per division.
         """
         dev = self.selectedDevice(c)
-        if scale is None:
-            resp = yield dev.query('CHAN%d:SCAL?' %channel)
-        else:
-            scale = format(scale,'E')
-            yield dev.write(('CHAN%d:SCAL '+scale) %channel)
-            resp = yield dev.query('CHAN%d:SCAL?' %channel)
+        if scale is not None:
+            yield dev.write('CHAN{}:SCAL {:E}'.format(channel, scale))
+        resp = yield dev.query('CHAN{}:SCAL?'.format(channel))
         scale = float(resp)
         returnValue(scale)
 
-    @setting(113, channel = 'i', factor = 'i', returns = ['s'])
-    def probeNONEXISTANT(self, c, channel, factor = None):
-        """Get or set the probe attenuation factor.
-        """
-        raise Exception('Not yet implemented')
-        probeFactors = [1,10,20,50,100,500,1000]
-        dev = self.selectedDevice(c)
-        chString = 'CH%d:' %channel
-        if factor is None:
-            resp = yield dev.query(chString+'PRO?')
-        elif factor in probeFactors:
-            yield dev.write(chString+'PRO %d' %factor)
-            resp = yield dev.query(chString+'PRO?')
-        else:
-            raise Exception('Probe attenuation factor not in '+str(probeFactors))
-        returnValue(resp)
-
-    @setting(114, channel = 'i', state = '?', returns = 's')
-    def channelOnOff(self, c, channel, state = None):
+    @setting(114, channel='i', state='?', returns='s')
```

Comment:
In fact, there's no reason to convert state to a string if it is an int, since it will still get formatted just fine. So can do:
```python
if state is not None:
    if isinstance(state, str):
        state = state.upper()
    if state not in [0, 1, 'ON', 'OFF']:
        raise Exception(...)
    yield dev.write(...)
```